tests/integration/test_authd/test_remote_enrollment.py

```diff
@@ -0,0 +1,196 @@
+
+targets:
+    - manager
+
+daemons:
+    - wazuh-authd
+    - wazuh-db
+    - wazuh-modulesd
+
+os_platform:
+    - linux
+
+os_version:
+    - Arch Linux
+    - Amazon Linux 2
+    - Amazon Linux 1
+    - CentOS 8
+    - CentOS 7
+    - Debian Buster
+    - Red Hat 8
+    - Ubuntu Focal
+    - Ubuntu Bionic
+
+references:
+    - https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/auth.html#remote-enrollment
+
+tags:
+    - enrollment
+'''
+import os
+import pytest
+import socket, time
+from pathlib import Path
+
+from wazuh_testing.constants.paths.logs import WAZUH_LOG_PATH
+from wazuh_testing.utils import callbacks
+from wazuh_testing.modules.authd import PREFIX
+from wazuh_testing.tools.socket_controller import SocketController
+from wazuh_testing.tools.monitors import file_monitor
+from wazuh_testing.utils.wazuh import DEFAULT_SSL_REMOTE_ENROLLMENT_PORT
+from wazuh_testing.utils.configuration import load_configuration_template, get_test_cases_data
+from contextlib import contextmanager
+
+from . import CONFIGURATIONS_FOLDER_PATH, TEST_CASES_FOLDER_PATH
+
+# Marks
+pytestmark = [pytest.mark.linux, pytest.mark.tier(level=0), pytest.mark.server]
+
+
+# Configurations
+test_configuration_path = Path(CONFIGURATIONS_FOLDER_PATH, 'config_remote_enrollment.yaml')
+test_cases_path = Path(TEST_CASES_FOLDER_PATH, 'cases_remote_enrollment.yaml')
+test_configuration, test_metadata, test_cases_ids = get_test_cases_data(test_cases_path)
+test_configuration = load_configuration_template(test_configuration_path, test_configuration, test_metadata)
+
+# Variables
+log_monitor_paths = []
+
+receiver_sockets_params = [(("localhost", DEFAULT_SSL_REMOTE_ENROLLMENT_PORT), 'AF_INET', 'SSL_TLSv1_2')]
+
+monitored_sockets_params = [('wazuh-modulesd', None, True), ('wazuh-db', None, True), ('wazuh-authd', None, True)]
+
+receiver_sockets, monitored_sockets = None, None  # Set in the fixtures
+
+cluster_socket_address = ('localhost', 1516)
+remote_enrollment_address = ('localhost', 1515)
+
+AGENT_ID = 0
+AGENT_NAME = 'test_agent'
+INPUT_MESSAGE = "OSSEC A:'{}_{}'"
+
+
+def wait_for_tcp_port(port, host='localhost', timeout=10):
+    """Wait until a port starts accepting TCP connections.
+    Args:
+        port (int): Port number.
+        host (str): Host address on which the port should be listening. Default 'localhost'
+        timeout (float): In seconds. How long to wait before raising errors.
+    Raises:
+        TimeoutError: The port isn't accepting connection after time specified in `timeout`.
+    """
+    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    start_time = time.time()
+    while time.time() - start_time < timeout:
+        try:
+            sock.connect((host, port))
+            sock.close()
+            return
+        except ConnectionRefusedError:
+            time.sleep(1)
+
+
+    raise TimeoutError(f'Waited too long for the port {port} on host {host} to start accepting messages')
+
+@contextmanager
```

Comment:
What is this for?

Reply:
I searched for it and found the related PR: https://github.com/wazuh/wazuh-qa/pull/1399.
Since the Python version has been updated, I can roll back that change